actioncable/lib/action_cable/connection/base.rb

```diff
@@ -224,6 +224,11 @@ def new_tagged_logger
             tags: server.config.log_tags.map { |tag| tag.respond_to?(:call) ? tag.call(request) : tag.to_s.camelize }
         end
 
+        def websocket_create(env, event_target, event_loop, protocols: ActionCable::INTERNAL[:protocols])
+          return ActionCable::Connection::WebSocketRack.new(env, event_target, event_loop, protocols) if ActionCable::Connection::WebSocketRack.okay?(env)
+          ActionCable::Connection::WebSocket.new(env, event_target, event_loop)
```

Comment:
What do you think about hiding this detail within `WebSocketRack` itself?

```ruby
if okay?
  return new
else
  return Websocket
```

that way, the healthy//unhealthy behavior is a private implementation detail of `WebSocketRack` vs a responsibility of the caller.

Reply:
Sure, we can do that. I'll rename the method name to `attempt`, so it doesn't appear as a boolean method.